### PR TITLE
app-benchmarks/sysbench: add snapshot

### PR DESCRIPTION
**Pull request by hydrapolic**
No recent releases upstream so far.

https://bugs.gentoo.org/show_bug.cgi?id=482864

Package-Manager: portage-2.3.1

### CODE REVIEW

**Comment by gktrk** [comment]
Are you interested in proxy-maintaining this? The metadata looks empty.

**Comment by hydrapolic** [comment]
@gktrk, yes, I'll try to maintain sysbench.

**Comment by gktrk** [comment]
@gentoo/proxy-maint

**Comment by gktrk** [comment]
I will modify your second commit to include proxy-maint in the metadata. Unfortunately, can't leave you alone in there yet :)

File: app-benchmarks/sysbench/sysbench-1.0_pre20161017.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	)
+
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	default
+
+	insinto /usr/share/${PN}/tests/db
+	doins sysbench/tests/db/*.lua || die
```
Comment (gktrk) [inline app-benchmarks/sysbench/sysbench-1.0_pre20161017.ebuild]: Should we worry about these:

```
checking whether to compile with Drizzle support... (cached) yes
checking whether to compile with libattachsql support... (cached) yes
checking whether to compile with Oracle support... (cached) no
checking whether to compile with PostgreSQL support... (cached) no
checking whether to compile with Lua support... (cached) yes
checking whether SHM_HUGETLB is declared... yes
checking for xsltproc... xsltproc
checking whether xsltproc works... yes
```

Especially, xsltproc seems like a non-optional build-time dependency.

**Comment by hydrapolic** [comment]
Thanks @gktrk, I'll do some more tests on this.

**Comment by hydrapolic** [comment]
@gktrk, I've pushed the changes. I'm not sure if it's possible to state that for either mysql or postgresql use flag you need lua.

**Comment by gktrk** [comment]
I've been reading the sources and `dev-util/cram` is needed for testing which is a python package. We may need to inherit a python eclass after all.

**Comment by hydrapolic** [comment]
@gktrk, added tests. Why would we need the python eclass?

**Comment by gktrk** [comment]
Also, can you squash the commit for tests into the one that adds the snapshot?

File: app-benchmarks/sysbench/sysbench-1.0_pre20161017.ebuild [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+CDEPEND="aio? ( dev-libs/libaio )
+	lua? ( dev-lang/lua:= )
+	mysql? ( virtual/libmysqlclient )
+	postgres? ( dev-db/postgresql:= )"
+
+DEPEND="${CDEPEND}
+	sys-devel/libtool:=
+	dev-libs/libxslt
+	test? ( dev-util/cram )"
```
Comment (hydrapolic) [inline app-benchmarks/sysbench/sysbench-1.0_pre20161017.ebuild]: For what reason?

File: app-benchmarks/sysbench/sysbench-1.0_pre20161017.ebuild [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+RDEPEND="${CDEPEND}"
+
+REQUIRED_USE="
+	mysql? ( lua )
+	postgres? ( lua )"
```
Comment (hydrapolic) [inline app-benchmarks/sysbench/sysbench-1.0_pre20161017.ebuild]: The database tests are written in Lua (https://github.com/akopytov/sysbench/tree/1.0/sysbench/tests/db) and without Lua support you cannot run them.

File: app-benchmarks/sysbench/sysbench-1.0_pre20161017.ebuild [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="6"
+
```
Comment (hydrapolic) [inline app-benchmarks/sysbench/sysbench-1.0_pre20161017.ebuild]: vcs-snapshot was also recommended by @pacho2, but reading the eclass:

# Note: this eclass is no longer needed with the new-style 'archive'
# GitHub URLs. They have sane directory names and stable contents,
# so you should really prefer them.

The example code shows usage with https://github.com/example/${PN}/tarball/v${PV} -> ${P}.tar.gz link which is meant for releases.

**Comment by hydrapolic** [comment]
@gktrk, pushed.

**Comment by hydrapolic** [comment]
Thanks @gktrk for the feedback / tips.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/commit/7d2f5e8cda89ab5ee201ab654cf5ce3358223781. Thanks!

**Pull request closed without merging**